src/__swaps__/screens/Swap/utils/assets.ts

```diff
@@ -0,0 +1,347 @@
+import { AddressZero } from '@ethersproject/constants';
+import isValidDomain from 'is-valid-domain';
+
+import { ETH_ADDRESS, SupportedCurrencyKey } from '@/references';
+import {
+  AddressOrEth,
+  AssetApiResponse,
+  ParsedAsset,
+  ParsedSearchAsset,
+  ParsedUserAsset,
+  UniqueId,
+  ZerionAsset,
+  ZerionAssetPrice,
+} from '@/__swaps__/screens/Swap/types/assets';
+import { ChainId, ChainName } from '@/__swaps__/screens/Swap/types/chains';
+
+import { metadataClient } from '@/graphql';
+import * as i18n from '@/languages';
+import { SearchAsset } from '../types/search';
+
+import { chainIdFromChainName, chainNameFromChainId, customChainIdsToAssetNames, isNativeAsset } from './chains';
+import {
+  convertAmountAndPriceToNativeDisplay,
+  convertAmountToBalanceDisplay,
+  convertAmountToNativeDisplay,
+  convertAmountToPercentageDisplay,
+  convertAmountToRawAmount,
+  convertRawAmountToDecimalFormat,
+} from './numbers';
+import { Token } from '@/graphql/__generated__/metadata';
+
+const get24HrChange = (priceData?: ZerionAssetPrice) => {
+  const twentyFourHrChange = priceData?.relative_change_24h;
+  return twentyFourHrChange ? convertAmountToPercentageDisplay(twentyFourHrChange) : '';
+};
+
+export const getCustomChainIconUrl = (chainId: ChainId, address: AddressOrEth) => {
+  if (!chainId || !customChainIdsToAssetNames[chainId]) return '';
+  const baseUrl = 'https://raw.githubusercontent.com/rainbow-me/assets/master/blockchains/';
```

Comment:
worth looking into this, p sure we should be using cloudinary url as the base, q for @welps

Reply:
Yes! you can use `https://rainbowme-res.cloudinary.com/image/upload/v1693266228/assets` as baseUrl. RN the blockchains that are being uploaded to cloudinary are ([defined here](https://github.com/rainbow-me/assets/blob/31d652aaf3a348c23303e4402993029e8b40279e/internal/service/staging_assets_service.go#L40)):
* ETHEREUM: ethereum
* POLYGON: polygon
* OPTIMISM: optimism
* ARBITRUM: arbitrum
* SMARTCHAIN: smartchain
* BASE: base
* AVALANCHEC: avalanchec

For example for avax (zero address on avalanche), 
* the image link from github would be: `https://github.com/rainbow-me/assets/blob/master/blockchains/avalanchec/assets/0x0000000000000000000000000000000000000000/logo.png`
* The same image, under the CDN (in order to be cached), would be `https://rainbowme-res.cloudinary.com/image/upload/v1693266228/assets/avalanchec/0x0000000000000000000000000000000000000000.png`